src/components/ImageBackground.js

```diff
@@ -11,19 +11,52 @@ type OwnProps = {
   reference: FieldRef
 };
 
+type State = {
+  imageSize: ?{ width: number, height: number }
+};
+
 type StateProps = {
-  getImageDetails: string => ?ImageDetails
+  getImageDetails: (string, { width: number, height: number }) => ?ImageDetails
 };
 
 type Props = OwnProps & StateProps;
 
-export const ImageBackground = ({
-  reference,
-  getImageDetails,
-  ...props
-}: Props) => (
-  <RNImageBackground source={getImageDetails(reference.sys.id)} {...props} />
-);
+export class ImageBackground extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { imageSize: null };
+  }
+
+  onLayout = (event: Object) => {
```

Comment:
React Native exposes the flow type for the layout event. You could use this here.

Example:

https://github.com/redbadger/pride-london-app/blob/1b40ff6b61b40e00eaacc0a87daa59b220a2325d/src/components/Collapsible.js#L7

https://github.com/redbadger/pride-london-app/blob/1b40ff6b61b40e00eaacc0a87daa59b220a2325d/src/components/Collapsible.js#L65-L67